lib/spack/spack/build_systems/cmake.py

```diff
@@ -459,8 +459,15 @@ def cmake(
         """Runs ``cmake`` in the build directory"""
 
         # skip cmake phase if it is an incremental develop build
+        # These are the files that will re-run CMake that are generated from a successful
+        # configure step
+        if self.generator == "Unix Makefiles":
+            configure_artifact = "Makefile"
+        elif self.generator == "Ninja":
```

Comment:
```suggestion
        # configure step
        primary_generator = _extract_primary_generator(self.generator)
        if primary_generator == "Unix Makefiles":
            configure_artifact = "Makefile"
        elif primary_generator == "Ninja":
```
@greenc-FNAL This is what you wanted?